lib/src/test/java/com/basistheory/android/service/BasisTheoryElementsTests.kt

```diff
@@ -251,4 +254,193 @@ class BasisTheoryElementsTests {
 
             verify { tokenizeApi.tokenize(expectedRequest) }
         }
+
+    @Test
+    fun `createToken should pass api key override to ApiClientProvider`() = runBlocking {
+        val apiKeyOverride = UUID.randomUUID().toString()
+
+        every { provider.getTokensApi(any()) } returns tokensApi
+
+        bt.createToken(CreateTokenRequest(), apiKeyOverride)
+
+        verify { provider.getTokensApi(apiKeyOverride) }
+    }
+
+    @Test
+    fun `createToken should forward top level primitive value without modification`() =
+        runBlocking {
+            every { provider.getTokensApi(any()) } returns tokensApi
+
+            val name = faker.name().fullName()
+            val createTokenRequest = createTokenRequest(name)
+            bt.createToken(createTokenRequest)
+
+            verify { tokensApi.create(createTokenRequest) }
+        }
+
+    @Test
+    fun `createToken should forward complex data values within request without modification`() =
+        runBlocking {
+            every { provider.getTokensApi(any()) } returns tokensApi
+
+            val data = object {
+                val string = faker.lorem().word()
+                val int = faker.random().nextInt(10, 100)
+                val nullValue = null
+                val nested = object {
+                    val double = faker.random().nextDouble()
+                    val bool = faker.random().nextBoolean()
+                    val timestamp = Instant.now().toString()
+                    val nullValue = null
+                }
+            }
+            val request = createTokenRequest(data)
+
+            bt.createToken(request)
```

Comment:
There has to be a way to write these tests once and run them against both createToken and tokenize... let's think about refactoring this next time we're in here.